app/features/quizzify/tools.py

```diff
@@ -317,9 +334,12 @@ def create_questions(self, num_questions: int = 5) -> List[Dict]:
 
         while len(generated_questions) < num_questions and attempts < max_attempts:
             response = chain.invoke(self.topic)
+
+            response = transform_json_dict(response)
+
             if self.verbose:
                 logger.info(f"Generated response attempt {attempts + 1}: {response}")
-            
+
             # Directly check if the response format is valid
             if self.validate_response(response):
```

Comment:
Like right here we validate the response AFTER transforming the dictionary. If the LLM hallucinates, we will break in the function too early. So `transform_json_dict` should try first to transform the output and keep validate response since that is part of our while loop

Reply:
@mikhailocampo  Yes, `transform_json_dict` first transforms the output if it actually has the different key/value structure, which is:
![image](https://github.com/radicalxdev/kai-ai-backend/assets/113047749/33fca3ff-a5db-4fbb-ab2b-bcb6904be195)
If the provided structures is not like the one that I shared before, it doesn't perform the transformation.

Reply:
So, when it has the incorrect structure, the function adapts it to the appropriate one so that `validate_response` can be appropriately used

Reply:
For example, as you can see, when the function is applied, it only needs 5 requests for 5 questions: 
![image](https://github.com/radicalxdev/kai-ai-backend/assets/113047749/e965528b-b7c5-495b-b09e-1566967c022c)
And the questions are correctly formatted: 
![image](https://github.com/radicalxdev/kai-ai-backend/assets/113047749/ab7a18e8-ec99-483a-9070-c2c03320fb1a)

Reply:
But when we don't use the function, it generally uses 8 to 10 requests for generating only 5 questions:
![image](https://github.com/radicalxdev/kai-ai-backend/assets/113047749/a91898e3-f430-45de-a2ee-f0b3fa8803dd)

Reply:
As you have seen, the `transform_json_dict` changes the response only when it doesn't have the accepted format. Therefore, when we used the function, there were no retries or incorrect formats provided to the model.

Reply:
Using the function as an intermediate process before of validating the dict enhances the rate of successful validation.

Reply:
Yes, I've already refactored the transform_json_dict as you requested @mikhailocampo